src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/domain/repository/OfferingRepository.kt

```diff
@@ -9,6 +11,9 @@ import java.util.UUID
 interface OfferingRepository : JpaRepository<OfferingEntity, UUID> {
   fun findAllByCourseId(courseId: UUID): List<OfferingEntity>
   fun findByCourseIdAndOrganisationIdAndWithdrawnIsFalse(courseId: UUID, organisationId: String): OfferingEntity?
-
   fun findByCourseIdAndIdAndWithdrawnIsFalse(courseId: UUID, offeringId: UUID): OfferingEntity?
+
+  @Modifying
+  @Query("delete from OfferingEntity o where o.id = :offeringId")
+  fun delete(offeringId: UUID)
```

Comment:
any reason for not using `offeringRepository.deleteById()` ?

Reply:
yeah I tried that too and it doesn't work. Burned too much time looking into it so went with that option. Maybe one day I'll work out why this entity is different.